### PR TITLE
sys-fs/zfs-auto-snapshot: new package

### PR DESCRIPTION
**Pull request by gyakovlev**
Automatically create, rotate, and destroy periodic ZFS snapshots.

Current spread: https://repology.org/metapackage/zfs-auto-snapshot/versions
Closes: https://bugs.gentoo.org/437636

### CODE REVIEW

**Comment by gyakovlev** [comment]
found minor issue with live ebuild, force-fushed.
ping, let's review it and get merged.

File: sys-fs/zfs-auto-snapshot/zfs-auto-snapshot-1.2.4.ebuild [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+if [[ ${PV} == 9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/zfsonlinux/${PN}.git"
+else
+	SRC_URI="https://github.com/zfsonlinux/${PN}/archive/upstream/${PV}.tar.gz -> ${P}.tar.gz"
```
Comment (mgorny) [inline sys-fs/zfs-auto-snapshot/zfs-auto-snapshot-1.2.4.ebuild]: Could you please use the same distfile name as the directory inside it? i.e. `${PN}-upstream-${PV}`. You can also put that string in common variable `MY_P` for reuse.

File: sys-fs/zfs-auto-snapshot/zfs-auto-snapshot-1.2.4.ebuild [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	KEYWORDS="~amd64"
+	S="${WORKDIR}/${PN}-upstream-${PV}"
+fi
+
+DESCRIPTION="ZFS Automatic Snapshot Service for Linux"
+HOMEPAGE="https://github.com/zfsonlinux/zfs-auto-snapshot"
+
+LICENSE="GPL-2"
```
Comment (mgorny) [inline sys-fs/zfs-auto-snapshot/zfs-auto-snapshot-1.2.4.ebuild]: `GPL-2+`

File: sys-fs/zfs-auto-snapshot/zfs-auto-snapshot-1.2.4.ebuild [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+SLOT="0"
+IUSE="+default-exclude"
+
+RDEPEND="sys-fs/zfs
+		virtual/cron"
```
Comment (mgorny) [inline sys-fs/zfs-auto-snapshot/zfs-auto-snapshot-1.2.4.ebuild]: One tab for indent, please.

File: sys-fs/zfs-auto-snapshot/zfs-auto-snapshot-1.2.4.ebuild [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+src_install() {
+	if use default-exclude; then
+		for cronfile in etc/${PN}.cron.{daily,hourly,monthly,weekly}; do
+			sed -i "s/\(^exec ${PN}\)/\1 --default-exclude/" "$cronfile" || die
```
Comment (mgorny) [inline sys-fs/zfs-auto-snapshot/zfs-auto-snapshot-1.2.4.ebuild]: Always use `${...}` form in ebuilds.

File: sys-fs/zfs-auto-snapshot/zfs-auto-snapshot-1.2.4.ebuild [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+		done
+		sed -i "s/\(; ${PN}\)/\1 --default-exclude/" etc/"${PN}".cron.frequent
+	fi
+	emake PREFIX="${EPREFIX}/usr" DESTDIR="${D}" install
+}
+
+pkg_postinst() {
+	if ! use default-exclude; then
```
Comment (mgorny) [inline sys-fs/zfs-auto-snapshot/zfs-auto-snapshot-1.2.4.ebuild]: Optional suggestion: use `readme.gentoo-r1` eclass to additional store that in `README.gentoo` for future read.

**Comment by gyakovlev** [comment]
implemented all suggestions

**Comment by mgorny** [comment]
Thanks a lot!

**Pull request closed without merging**